integrations/hubspot/tests/hubspot-contacts.test.ts

```diff
@@ -0,0 +1,52 @@
+import { vi, expect, it, describe } from 'vitest';
+
+import fetchData from '../syncs/contacts.js';
+
+describe('hubspot contacts tests', () => {
+    const nangoMock = new global.vitest.NangoSyncMock({
+        dirname: __dirname,
+        name: 'contacts',
+        Model: 'HubspotContact'
+    });
+
+    const models = 'HubspotContact'.split(',');
+    const batchSaveSpy = vi.spyOn(nangoMock, 'batchSave');
+
+    it('should get, map correctly the data and batchSave the result', async () => {
+        await fetchData(nangoMock);
+
+        for (const model of models) {
+            const batchSaveData = await nangoMock.getBatchSaveData(model);
+
+            const totalCalls = batchSaveSpy.mock.calls.length;
+
+            if (totalCalls > models.length) {
+                const splitSize = Math.ceil(batchSaveData.length / totalCalls);
+
+                const splitBatchSaveData = [];
+                for (let i = 0; i < totalCalls; i++) {
+                    const chunk = batchSaveData.slice(i * splitSize, (i + 1) * splitSize);
```

Comment:
this could be simplified to:
```
const chunk = batchSaveData.slice(i, i + chunkSize)
```
if the for loop becomes
```
for (let i = 0; i < batchSaveData.length; i += splitSize) {
```

Reply:
Noted, will improve at a later iteration. Thanks.